### ISSUE TITLE
I can't download the model after training in the example

### ISSUE THREAD

**Issue by xuanco941** [keras-team/keras-io#1364]
link example : [](https://github.com/keras-team/keras-io/blob/master/examples/vision/image_captioning.py
![image](https://github.com/keras-team/keras-io/assets/68849866/46b59339-3736-40e9-a912-b0ddb0a7f90f)

error in line caption_model.save().
my error: Model <main.ImageCaptioningModel object at 0x7f0ea05a40d0> cannot be saved either because the input shape is not available or because the forward pass of the model is not defined.To define a forward pass, please override Model.call(). To specify an input shape, either call build(input_shape) directly, or call the model on actual data using Model(), Model.fit(), or Model.predict(). If you have a custom training step, please make sure to invoke the forward pass in train step through Model.__call__, i.e. model(inputs), as opposed to model.call().

**Comment by sachinprasadhs** [comment]
I was able to reproduce the issue with Tensorflow 2.12 version, please find the attached Gist [here](https://gist.github.com/sachinprasadhs/09a5699fc5571e5cb4dba43f61901af1). Thanks!

**Comment by rchao** [comment]
@AakashKumarNain would you be able to help with this issue? Thanks!

**Comment by AakashKumarNain** [comment]
Can you try saving the weights instead?

**Comment by shaunaa126** [comment]
I was able to save the weights using this command `caption_model.save_weights(SAVE_DIR + 'model_weights_flickr8k.h5')`

The benefit of [Saved_model](https://www.tensorflow.org/guide/keras/serialization_and_saving) is to be able to save the entire model including weights, models, model architecture into a folder... which can be later converted to tflite model for inference. However you are suggestion to save just the weights. How can we restore the model and restore the weights after training?

**Comment by AakashKumarNain** [comment]
I am aware of that. The problem is that we would be required to override `get_config`, and `from_config` methods, and we would need to register the custom objects. You can try doing that but I am not sure if that would work for layers with a `mask` argument
